x/upgrade/spec/01_concepts.md

```diff
@@ -56,6 +56,17 @@ During each `EndBlock` execution, the `x/upgrade` module checks if there exists
 `Handler` is executed. If the `Plan` is expected to execute but no `Handler` is registered
 or if the binary was upgraded too early, the node will gracefully panic and exit.
 
+## StoreLoader
+The `x/upgrade` module also facilitates store migrations as part of the upgrade. The
```

Comment:
I think we certainly need way more detail and explanation here. This is a core component of the upgrade procedure. I'd elaborate and explain concepts and procedures such as when and how a file is written to disk. What the file contains and why. How the file is used to migrate state....etc.